aptos-move/framework/aptos-framework/sources/dora_committee.move

```diff
@@ -0,0 +1,1042 @@
+        network_public_key: vector<vector<u8>>,
+        elgamal_pub_key: vector<vector<u8>>,
+        network_port: vector<u16>,
+        rpc_port: vector<u16>,
+        committee_type: u8
+    ) acquires CommitteeInfoStore, SupraCommitteeEventHandler {
+        // Assert the length of the vector for two are the same
+        let node_address_len = vector::length(&node_addresses);
+        assert!(
+            node_address_len == vector::length(&ip_public_address),
+            error::invalid_argument(INVALID_COMMITTEE_NUMBERS)
+        );
+        assert!(
+            node_address_len == vector::length(&dora_public_key),
+            error::invalid_argument(INVALID_COMMITTEE_NUMBERS)
+        );
+        assert!(
+            node_address_len == vector::length(&network_public_key),
+            error::invalid_argument(INVALID_COMMITTEE_NUMBERS)
+        );
+        assert!(
+            node_address_len == vector::length(&elgamal_pub_key),
+            error::invalid_argument(INVALID_COMMITTEE_NUMBERS)
+        );
+        assert!(
+            node_address_len == vector::length(&network_port),
+            error::invalid_argument(INVALID_COMMITTEE_NUMBERS)
+        );
+        assert!(
+            node_address_len == vector::length(&rpc_port),
+            error::invalid_argument(INVALID_COMMITTEE_NUMBERS)
+        );
+        // Only the OwnerCap capability can access it
+        let _acquire = &capability::acquire(owner_signer, &OwnerCap {});
+        let committee_store = borrow_global_mut<CommitteeInfoStore>(com_store_addr);
+        let dora_node_info = vector::empty<DoraNodeInfo>();
+        let node_addresses_for_iteration = node_addresses;
+        while (vector::length(&node_addresses_for_iteration) > 0) {
+            let ip_public_address = vector::pop_back(&mut ip_public_address);
+            let dora_public_key = vector::pop_back(&mut dora_public_key);
+            let network_public_key = vector::pop_back(&mut network_public_key);
+            let elgamal_pub_key = vector::pop_back(&mut elgamal_pub_key);
+            let network_port = vector::pop_back(&mut network_port);
+            let rpc_port = vector::pop_back(&mut rpc_port);
+            let dora_node = DoraNodeInfo {
+                ip_public_address: copy ip_public_address,
+                dora_public_key: copy dora_public_key,
+                network_public_key: copy network_public_key,
+                elgamal_pub_key: copy elgamal_pub_key,
+                network_port,
+                rpc_port,
+            };
+            vector::push_back(&mut dora_node_info, dora_node);
+            // Also update the node_to_committee_map
+            let node_address = vector::pop_back(&mut node_addresses_for_iteration);
+            simple_map::upsert(&mut committee_store.node_to_committee_map, node_address, id);
+        };
+        let committee_info = CommitteeInfo {
+            map: simple_map::new_from(node_addresses, dora_node_info),
```

Comment:
we need to reverse the DORA node info because we need to add records in sequential order with node addresses.
`vector::reverse(&mut dora_node_info);`

Reply:
Oh, you are right...Thank you

Reply:
Updated in the new PR. Thanks